addons/account/models/account_move.py

```diff
@@ -1321,36 +1321,122 @@ def _compute_needed_terms(self):
                     }
 
     def _compute_payments_widget_to_reconcile_info(self):
-        for move in self:
-            move.invoice_outstanding_credits_debits_widget = False
-            move.invoice_has_outstanding = False
-
-            if move.state not in {'draft', 'posted'} \
-                    or move.payment_state not in ('not_paid', 'partial') \
-                    or not move.is_invoice(include_receipts=True):
-                continue
-
-            pay_term_lines = move.line_ids\
-                .filtered(lambda line: line.account_id.account_type in ('asset_receivable', 'liability_payable'))
-
-            domain = [
-                ('account_id', 'in', pay_term_lines.account_id.ids),
+        """ Compute outstanding payments information in bulk """
+        if not self:
+            return
+        
+        def _get_common_domain(commercial_partner_id):
+            """ Return the common domain conditions for searching outstanding payments """
+            return [
                 ('parent_state', '=', 'posted'),
-                ('partner_id', '=', move.commercial_partner_id.id),
+                ('partner_id', '=', commercial_partner_id),
                 ('reconciled', '=', False),
                 '|', ('amount_residual', '!=', 0.0), ('amount_residual_currency', '!=', 0.0),
             ]
 
-            payments_widget_vals = {'outstanding': True, 'content': [], 'move_id': move.id}
+        moves_to_process = self.filtered(
+            lambda m: m.state in {'draft', 'posted'} 
+            and m.payment_state in ('not_paid', 'partial') 
+            and m.is_invoice(include_receipts=True)
+        )
+        
+        for move in self:
+            move.invoice_outstanding_credits_debits_widget = False
+            move.invoice_has_outstanding = False
+        
+        if not moves_to_process:
+            return
 
+        company_ids = moves_to_process.mapped('company_id')
```

Comment:
Why do you declare the variable here? Also for relational fields, you don't need to call `mapped`.
Seems like you only use it once, you could just do 
```python
for company in moves_to_process.company_id:
```